lib/push.js

```diff
@@ -35,25 +44,31 @@ module.exports = function (log, db) {
             if (device.pushCallback) {
               // send the push notification
               log.increment(LOG_UPDATE_INCREMENT_SEND)
-              request.post(device.pushCallback, function (err){
+              request.post(device.pushCallback, function (err, response) {
                 if (err) {
-                  log.error({
-                    op: LOG_OP_NOTIFY_UPDATE,
-                    deviceId: deviceId,
-                    err: err
-                  })
-                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                  // any 400 level error from the push servers means
+                  // the push settings need to be reset.
+                  // the clients will check this and re-register push endpoints
+                  if (response && response.statusCode >= 400 && response.statusCode < 500) {
+                    // reset device push configuration
+                    // Warning: this method is called without any session tokens or auth validation.
+                    device.pushCallback = ''
+                    device.pushPublicKey = ''
```

Comment:
@vladikoff, oh good point, it should be a 32-byte buffer, all zeroes I guess.

From a practical standpoint, as long as `pushCallback` and `pushPublicKey` are always get and set as a pair, there may be no harm in leaving the old public key hanging about and treating an empty callback field as sufficient signal that there is no push metadata. But for the sake of a neat and tidy database, zeroing it seems more proper I think.

@rfk or @dannycoates might want to chip in too at this point though, I'm still a back-end novice tbh.